lib/src/modules/task.cc

```diff
@@ -240,30 +239,43 @@ static std::string createUrlEndpoint(const lth_jc::JsonContainer& uri) {
 //    the same task file.
 // The downloaded task file's permissions will be set to rwx for user and rx for
 // group for non-Windows OSes.
-static void downloadTaskFile(const std::vector<std::string>& master_uris,
+static bool downloadTaskFile(const std::vector<std::string>& master_uris,
                              lth_curl::client& client,
                              const fs::path& file_path,
                              const lth_jc::JsonContainer& uri) {
-    auto url = master_uris[0] + createUrlEndpoint(uri);
-    lth_curl::request req(url);
-    // timeout from connection after one minute, can configure
-    req.connection_timeout(60000);
+    auto endpoint = createUrlEndpoint(uri);
+    for (auto& master_uri : master_uris) {
+        auto url = master_uri + endpoint;
+        lth_curl::request req(url);
+        // timeout from connection after one minute, can configure
+        req.connection_timeout(60000);
+        try {
 #ifndef _WIN32
-    client.download_file(req, file_path.string(), NIX_TASK_FILE_PERMS);
+            client.download_file(req, file_path.string(), NIX_TASK_FILE_PERMS);
 #else
-    client.download_file(req, file_path.string());
+            client.download_file(req, file_path.string());
 #endif
+            break;
+        } catch (lth_curl::http_file_download_exception& e) {
+            // Do nothing here -- we want to try the next master-uri.
+
+            // TODO: Should we collect all the exception messages here to pass in to the overall
+            // Module::ProcessingError if all master_uris fail? Might not be a bad idea.
```

Comment:
Writing to the disk sounds like a filesystem_error, while server stuff is a CURL error. Because I think we really write to the disk as in write to our main file at the last fs::rename part in download_file. But then there's the question of what if things fail at the write_file callback? According to https://curl.haxx.se/libcurl/c/libcurl-errors.html, a CURL-WRITE-ERROR is returned for that case. I think the question is "What would be considered a 'problem writing to disk' vs. connecting to the server?"